### PR TITLE
repair ui

### PR DESCRIPTION
**Pull request by stubbrn**
That's a start

### CODE REVIEW

**Comment by stubbrn** [comment]
From `stagenet.md`:

> Note: the swapd process directly interacts with an unlocked Monero wallet and Ethereum private key. This is to allow for a smoother swap process that doesn't require any interaction from you once initiated. However, this effectively gives swapd access to all your (testnet) funds. In the future, there will be a mode that does not access your keys/wallet, but will require user interaction during a swap.

However it seems like the front-end was working that way (i.e using metamask) at first, it was broken partly because of this, so tell me if what I did (Using `net_takeOfferAndSubscribe` instead of `signer_subscribe`) is adequate.
It's now supposed to be in a state of "barely working" again.

**Comment by noot** [comment]
@stubbrn thanks for this! will review shortly :)

File: ui/src/components/TakeDealDialog.svelte [hunk 1]
```diff
@@ -17,7 +17,6 @@
   const WS_ADDRESS = 'ws://127.0.0.1:5001/ws'
```
Comment (noot) [inline ui/src/components/TakeDealDialog.svelte:17]: can you change this to `ws://127.0.0.1:${SWAPD_PORT}/ws` where `SWAPD_PORT` is an environment variable? there's also a function for this `readSwapdPortNumber()` in utils/rpcApi.ts you may be able to use

Comment (noot) [reply ui/src/components/TakeDealDialog.svelte]: yeah it is! I think this was the only place it was still hard-coded, so this should fix #127 also

**Comment by noot** [comment]
overall, I think this is good as it un-breaks the UI, but I think it should still be using `signer_subscribe`, as it's a bit weird right now that it connects to metamask but doesn't sign anything with it. Is `signer_subscribe` completely broken? I could help with debugging there if that'd be helpful for you

**Review by dimalinux — APPROVED** [review]

**Pull request merged**